File: easy/1-bit and 2-bit Characters/Go/main.go
// Problem: 1-bit and 2-bit Characters
// Link to the problem: https://leetcode.com/problems/1-bit-and-2-bit-characters/
func isOneBitCharacter(bits []int) bool {
	var n int = len(bits)
	var i int = 0
	for i < n-1 {
		i += bits[i] + 1
	}
	return i == n-1
}

File: easy/1-bit and 2-bit Characters/Rust/main.rs
// Problem: 1-bit and 2-bit Characters
// Link to the problem: https://leetcode.com/problems/1-bit-and-2-bit-characters/
impl Solution {
    pub fn is_one_bit_character(bits: Vec<i32>) -> bool {
        let n: usize = bits.len();
        let mut i: usize = 0;
        while (i < n - 1) {
            i += (bits[i] + 1) as usize;
        }
        return i == n - 1;
    }
}
